plugins/lookup/validate.py

```diff
@@ -0,0 +1,167 @@
+          this case the value I(ansible.utils.jsonschema) represents the engine to be use for data valdiation.
+          If the value is not provided the default value that is I(ansible.uitls.jsonschema) will be used.
+        - The value should be in fully qualified collection name format that is
+          I(<org-name>.<collection-name>.<validate-plugin-name>).
+        default: ansible.utils.jsonschema
+    notes:
+    - For the type of options C(data) and C(criteria) refer the individual C(validate) plugin
+      documentation that is represented in the value of C(engine) option.
+    - For additional plugin configuration options refer the individual C(validate) plugin
+      documentation that is represented by the value of C(engine) option.
+    - The plugin configuration option can be either passed as I(key=value) pairs within lookup plugin
+      or task or environment variables.
+    - The precedence the C(validate) plugin configurable option is the variable passed within lookup plugin
+      as I(key=value) pairs followed by task variables followed by environment variables.
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    data: "{{ lookup('file', './validate/data/show_interfaces_iosxr.json')}}"
+    criteria: "{{ lookup('file', './validate/criteria/jsonschema/show_interfaces_iosxr.json')}}"
+
+- name: validate data in json format using jsonschema with lookup plugin by passing plugin configuration variable as key/value pairs
+  ansible.builtin.set_fact:
+    data_criteria_checks: "{{ lookup(data, criteria, engine='ansible.utils.jsonschema', draft='draft7') }}"
+
+- name: validate data in json format using jsonschema with lookup plugin by passing plugin configuration variable as task variable
+  ansible.builtin.set_fact:
+    data_criteria_checks: "{{ lookup('ansible.utils.validate', data, criteria, engine='ansible.utils.jsonschema', draft='draft7') }}"
+  vars:
+    ansible_validate_jsonschema_draft: draft3
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - If data is valid returns empty list
+      - If data is invalid returns list of errors in data
+"""
+
+from copy import deepcopy
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.plugins.lookup import LookupBase
+from ansible_collections.ansible.utils.plugins.module_utils.validate.base import (
+    load_validator,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    to_list,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+ARGSPEC_CONDITIONALS = {}
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+        if len(terms) < 2:
```

Comment:
did you see how I did this in the other lookups?  It's a nice pattern using zip

Reply:
Same as above